### ISSUE TITLE
console.html should limit size of output

### ISSUE THREAD

**Issue by hoodmane** [iodide-project/pyodide#1135]
The following code will hang the pyodide console:
```python
import numpy as np
arr = np.arange(1920*1080*3)
arr
```
We should use `reprlib.repr` with an appropriate size limit to format the output:
https://docs.python.org/3/library/reprlib.html#reprlib.repr

**Comment by hoodmane** [comment]
@casatir This may well be already in the list of improvements you are planning?

**Comment by casatir** [comment]
Thanks @hoodmane for catching this issue. Haven't noticed.

As you may know, in the second part of the console improvement I planed, I advertised in #1125 for a clean `repr` (already implemented in [improving-console-part-2](https://github.com/casatir/pyodide/tree/improving-console-part-2)). Just tested and in the precise case of a `numpy` array, the size limitting is correct:

```python
array([      0,       1,       2, ..., 6220797, 6220798, 6220799])
```

Of course, a `>>> [0] * 10 ** 5` will hang the console but CPython produces a full output for this too. If we want to mimic the CPython REPL behavior, it will be fine with the future improvement but if we want to avoid hangs from long repr result, it might be interesting to use `reprlib.repr` as you mentioned. WDYT?

**Comment by hoodmane** [comment]
I think we want to avoid hangs from long repr results. (Especially since the solution here is quite simple.) Nobody _wants_ the repl to hang when they accidentally write something that gives really long output, especially because in these circumstances the browser sometimes grabs up tons of ram and slows down the whole system.

**Comment by casatir** [comment]
This is implemented in #1141. Limiting to 1000 elements. Maybe too big?

**Comment by casatir** [comment]
Could we close this as it is fixed by #1141?
